### ISSUE TITLE
Add option to PlayPauseButton to scale button when sim is paused

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/scenery-phet#562]
A common request is to make the PlayPauseButton bigger when the sim is paused and the "play" button is shown. And I have found three occurrences of 
```js
    // make the play/pause button bigger when it is paused
    const pauseSizeIncreaseFactor = 1.25;
    model.isPlayingProperty.lazyLink( function( isPlaying ) {
      playPauseButton.scale( isPlaying ? ( 1 / pauseSizeIncreaseFactor ) : pauseSizeIncreaseFactor );
    } );
```
copied around so far.

I suggest an option for PlayPauseButton called pauseSizeIncreaseFactor that does this for us, default to 1.

This is required for issues like https://github.com/phetsims/energy-skate-park/issues/150/

**Comment by jessegreenberg** [comment]
`pauseSizeIncreaseFactor` is ambiguous, I think one may think that refers to the "pause" button rather than the paused state. How about `playButtonScaleFactor` instead?

**Comment by jessegreenberg** [comment]
Done in the above commit and used in energy-skate-park.

I see this being done still in 
- [ ] bending-light
- [ ] forces-and-motion-basics
- [ ] projectile-motion
- [ ] states-of-matter
- [ ] wave-on-a-string

**Comment by jessegreenberg** [comment]
This documentation is in SOMPlayPauseStepControl:
> The PhET convention is to do this for sims where interaction does NOT unpause the sim, which is true for all usages in this sim.

Thats helpful, im going to move it to PlayPauseButton.

**Comment by jessegreenberg** [comment]
Done in the above commits, used in sims where we are scaling up the "Play" button.

@ariel-phet can you please assign a developer to review? @samreid is marked as author of PlayPauseButton but I think anyone could review this change.

**Comment by samreid** [comment]
> A common request is to make the PlayPauseButton bigger when the sim is paused and the "play" button is shown.

To clarify, should this be an option?  Or should all simulations show the PlayPauseButton bigger when the sim is paused?

> How about playButtonScaleFactor instead?

Yes, that's a much clearer name.  But it sounds like it is a permanent scaling, should we also indicate that it is only the scale when paused?

Also a note to facilitate review-ability for changes like this.  It's helpful to have the commits related to the relevant issue.  In this case, the Bending Light and Forces and Motion: Basics commits were noted related to https://github.com/phetsims/scenery-phet/issues/298.  Projectile motion's commit message referred to "562", which refers to projectile-motion/issues/562, which does not exist.  States of matter and wave on a string were correctly linked to this issue.

Back to @jessegreenberg about those questions before further review.

**Comment by jessegreenberg** [comment]
Thanks for reviewing.

> To clarify, should this be an option? Or should all simulations show the PlayPauseButton bigger when the sim is paused?

I believe this should be an option. The current PhET convention for this is noted in https://github.com/phetsims/scenery-phet/issues/562#issuecomment-564780196 and now documented in PlayPauseButton

> should we also indicate that it is only the scale when paused?

I thought the name `playButtonScaleFactor` indicated that the scale factor applies to the "Play" button (when the sim is paused).

> Also a note to facilitate review-ability for changes like this

Sorry! That was my mistake, those commits should reference this issue and incorrect numbers were an accident.

**Comment by samreid** [comment]
How about renaming `playButtonScaleFactor` to `playButtonScaleFactorWhenPaused` so developers don't think it is a permanent scale factor, even when playing?

**Comment by jessegreenberg** [comment]
If we add `whenPaused` then `playButton` isn't necessary, how about `scaleFactorWhenPaused`, to reduce name length?

**Comment by samreid** [comment]
Right `scaleFactorWhenPaused` is the perfect name for a `PlayPauseButton` option.  I was confused thinking we were specifying an option for `TimeControlNode`, but I see it is already set up for nested options correctly.  Thanks for correcting me.  I still have more review to do here, but may not get to it for a few days.

**Comment by jessegreenberg** [comment]
OK great, thanks! The above commits rename the option `playButtonScaleFactor` to `scaleFactorWhenPaused`.

**Comment by samreid** [comment]
I reviewed the changes for this issue and tested in Wave on a String.  Everything looks reasonable and it works properly.  I saw potential for improvement by using `setScaleMagnitude` instead of repeated calls to `scale` and inverse `scale` and I committed that change.  Then I realized that would be incompatible with other external `scale` calls, so I reverted it.  But for that matter, the existing implementation is incompatible with `setScaleMagnitude` calls.  For instance:

```js
// pseudocode
const b = new PlayPauseButton({scaleFactorWhenPaused: 2, isPlaying: true}) // width is 100
b.isPlayingProperty.value = false; // width is 200
b.setScaleMagnitude(5,5); // width is 500
b.isPlayingProperty.value = false; // width is 250
```

To summarize, do we need to "shield" the `PlayPauseButton` so that it has full control over its own scale, and it cannot be put into an inconsistent state by a client (possibly via a private Node that does only scaling)?  Or at a very minimum document that clients should not put the scale in an inconsistent state, but that seems risky.

Back to @jessegreenberg for discussion, but also co-assigning @jonathanolson in case he has recommendations for this part.

**Comment by pixelzoom** [comment]
Raising priority to high.  Assigned to @jonathanolson and @jessegreenberg for 6+ months without forward progress.  How would you like to proceed?

**Comment by jessegreenberg** [comment]
I just reviewed https://github.com/phetsims/scenery-phet/issues/562#issuecomment-657966100 - If I understand the example in https://github.com/phetsims/scenery-phet/issues/562#issuecomment-657966100 I think it is behaving correctly. Is that last line supposed to be `b.isPlayingProperty.value = true;`? The option (renamed to `scaleFactorWhenNotPlaying`) is the scale factor relative to the "playing" state. So when the isPlayingProperty is true it should be half as large as when isPlayingProperty is false, as demonstrated in your example. Reassigning to @samreid.

**Comment by samreid** [comment]
I tried to create a breaking case, experimenting with scale changes like so:

```diff
Index: js/demo/ButtonsScreenView.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/demo/ButtonsScreenView.js b/js/demo/ButtonsScreenView.js
--- a/js/demo/ButtonsScreenView.js	(revision 384a70a205571990769465f83fce5c532b0e42fc)
+++ b/js/demo/ButtonsScreenView.js	(date 1614185678119)
@@ -145,7 +145,15 @@
     );
     toggleButtons.push( eyeButton );
 
-    const playPauseButton = new PlayPauseButton( toggleButtonProperties.isPlayingProperty );
+    const playPauseButton = new PlayPauseButton( toggleButtonProperties.isPlayingProperty, {
+      scaleFactorWhenNotPlaying: 2
+    } );
+
+    playPauseButton.scale( 2 );
+    toggleButtonProperties.isPlayingProperty.value = false;
+    playPauseButton.scale( 0.5 );
+    toggleButtonProperties.isPlayingProperty.value = true;
+
     toggleButtonProperties.isPlayingProperty.lazyLink(
       playing => console.log( 'PlayPauseButton pressed, playing=' + playing )
     );
```

But was unable to create the incorrect behavior.  My main concern was that roundoff error from ( * scale * 1/scale) repeatedly would accumulate, but that would be out in 10^-8 or lower.  I think this issue can be closed, thanks @jessegreenberg
